staging/src/k8s.io/client-go/plugin/pkg/client/auth/azure/README.md

```diff
@@ -0,0 +1,68 @@
+# Azure Active Directory plugin for client authentication
+
+This plugin provides an integration with [Azure CLI](https://github.com/Azure/azure-cli) and Azure Active Directory device flow.
+
+## Usage
+
+### Microsoft Azure CLI 2.0
+
+1. Configure the `apiserver` to use the Azure Active Directory as an OIDC provider with following options
+
+   ```
+   --oidc-client-id="https://management.core.windows.net/" \
+   --oidc-issuer-url="https://sts.windows.net/TENANT_ID/"
+   --oidc-username-claim="sub"
+   ```
+
+   * Replace `TENANT_ID` with your tenant ID.
+
+2. Login with Azure CLI
+
+  ```
+  az login
+  ```
+
+3. Configure the `kubectl` to use the `azure` authentication provider
+
+   ```
+   kubectl config set-credentials "USER_NAME" --auth-provider=azure
+   ```
+
+   * Replace `USER_NAME` with your user name.
+
+### Device Flow
+
+1. Create an Azure native application following these [instructions](https://docs.microsoft.com/en-us/azure/active-directory/active-directory-app-registration)
+
+  Assign permissions to this application to access the `https://management.core.windows.net/"` audience.
+
+2. Configure the `apiserver` to use the Azure Active Directory as an OIDC provider with following options
+
+   ```
+   --oidc-client-id="https://management.core.windows.net/" \
```

Comment:
I get you point a client needs to request a token for itself. In that case AAD will return:

```
{
  "aud": "spn:319e3b43-c214-43e8-8d8d-8d33b8e47902",
  "iss": "https://sts.windows.net/72f988bf-86f1-41af-91ab-2d7cd011db47/",
   "appid": "319e3b43-c214-43e8-8d8d-8d33b8e47902",
}
```
Are you fine if the oidc-id is configured to `spn:319e3b43-c214-43e8-8d8d-8d33b8e479021`?

@ericchiang @colemickens 
I guess, the azure cli integration has to be removed because it adhere to OAuth2 actors model (http://nordicapis.com/api-security-oauth-openid-connect-depth/) requesting always a token for a resource server not for itself. 

I can keep the device code and also remove the `audience` configuration.